### PR TITLE
[MRG+1] FIX: Fewer warnings

### PR DESCRIPTION
**Pull request by larsoner**
Trying to help get CircleCI working I get a ton of warnings on my local machine with `make html-strict`. This should greatly reduce the number of warnings.

I don't know the details of the examples so I don't know if the `solver`, `cv`, and `iid` params are correct but they at least adhere to what I think the defaults had been previously.

The `show` function that looks at the backend type is something we use over in MNE to avoid the nagging matplotlib warnings about using the `Agg` backend.

### CODE REVIEW

File: examples/02_decoding/plot_haxby_different_estimators.py [hunk 1]
```diff
@@ -69,16 +69,16 @@
 # multiple cores
 svm_cv = GridSearchCV(SVC(C=1., kernel="linear"),
                       param_grid={'C': [.1, .5, 1., 5., 10., 50., 100.]},
-                      scoring='f1', n_jobs=1)
+                      scoring='f1', n_jobs=1, cv=3, iid=True)
 
 logistic_cv = GridSearchCV(LogisticRegression(C=1., penalty="l1"),
                            param_grid={'C': [.1, .5, 1., 5., 10., 50., 100.]},
-                           scoring='f1')
+                           scoring='f1', cv=3, iid=True)
```
Comment (GaelVaroquaux) [inline examples/02_decoding/plot_haxby_different_estimators.py]: iid is deprecated, and will default to False. We shouldn't be putting it in an example

File: examples/02_decoding/plot_haxby_searchlight.py [hunk 2]
```diff
@@ -70,16 +70,20 @@
...
+import warnings
 # The radius is the one of the Searchlight sphere that will scan the volume
-searchlight = nilearn.decoding.SearchLight(
-    mask_img,
-    process_mask_img=process_mask_img,
-    radius=5.6, n_jobs=n_jobs,
-    verbose=1, cv=cv)
-searchlight.fit(fmri_img, y)
+with warnings.catch_warnings():
+    warnings.simplefilter('ignore', ConvergenceWarning)  # might not converge
```
Comment (GaelVaroquaux) [inline examples/02_decoding/plot_haxby_searchlight.py]: I am in favor of putting this inside the code of searchlight.

**Comment by GaelVaroquaux** [comment]
Thanks you @larsoner . This is so useful!!

+1 for merge once CI passes, and after addressing my small comments.

@KamalakerDadi : do check that there is no overlap with some of your current efforts.

File: examples/02_decoding/plot_haxby_different_estimators.py (hunk 1 above)
Comment (larsoner) [reply examples/02_decoding/plot_haxby_different_estimators.py]: Okay, I'll set it to `False`

**Comment by KamalakerDadi** [comment]
>@KamalakerDadi : do check that there is no overlap with some of your current efforts.

Nope. @larsoner PR is good. The only thing I want to check is the use of CV. The results might change. Have to wait and see with the build.

**Comment by GaelVaroquaux** [comment]
I wouldn't worry about minor things that change for different values of CV.

**Comment by larsoner** [comment]
@GaelVaroquaux done

**Comment by GaelVaroquaux** [comment]
LGTM. +1 for merge once CI has ran.

**Comment by KamalakerDadi** [comment]
Seems like we have failure with the example: 02_decoding/plot_haxby_different_estimators.py

```python
/home/circleci/miniconda3/envs/testenv/lib/python3.6/site-packages/sklearn/model_selection/_validation.py:542: FutureWarning: From version 0.22, errors during fit will result in a cross validation score of NaN by default. Use error_score='raise' if you want an exception raised or error_score=np.nan to adopt the behavior from version 0.22.
  FutureWarning)
/home/circleci/project/doc/sphinxext/sphinx_gallery/gen_rst.py:491: UserWarning: ________________________________________________________________________________
/home/circleci/project/examples/02_decoding/plot_haxby_different_estimators.py failed to execute correctly:Traceback (most recent call last):
  File "/home/circleci/project/doc/sphinxext/sphinx_gallery/gen_rst.py", line 472, in execute_code_block
    exec(code_block, example_globals)
  File "<string>", line 23, in <module>
  File "/home/circleci/miniconda3/envs/testenv/lib/python3.6/site-packages/sklearn/model_selection/_validation.py", line 402, in cross_val_score
    error_score=error_score)
  File "/home/circleci/miniconda3/envs/testenv/lib/python3.6/site-packages/sklearn/model_selection/_validation.py", line 240, in cross_validate
    for train, test in cv.split(X, y, groups))
  File "/home/circleci/miniconda3/envs/testenv/lib/python3.6/site-packages/sklearn/externals/joblib/parallel.py", line 983, in __call__
    if self.dispatch_one_batch(iterator):
  File "/home/circleci/miniconda3/envs/testenv/lib/python3.6/site-packages/sklearn/externals/joblib/parallel.py", line 825, in dispatch_one_batch
    self._dispatch(tasks)
  File "/home/circleci/miniconda3/envs/testenv/lib/python3.6/site-packages/sklearn/externals/joblib/parallel.py", line 782, in _dispatch
    job = self._backend.apply_async(batch, callback=cb)
  File "/home/circleci/miniconda3/envs/testenv/lib/python3.6/site-packages/sklearn/externals/joblib/_parallel_backends.py", line 182, in apply_async
    result = ImmediateResult(func)
  File "/home/circleci/miniconda3/envs/testenv/lib/python3.6/site-packages/sklearn/externals/joblib/_parallel_backends.py", line 545, in __init__
    self.results = batch()
  File "/home/circleci/miniconda3/envs/testenv/lib/python3.6/site-packages/sklearn/externals/joblib/parallel.py", line 261, in __call__
    for func, args, kwargs in self.items]
  File "/home/circleci/miniconda3/envs/testenv/lib/python3.6/site-packages/sklearn/externals/joblib/parallel.py", line 261, in <listcomp>
    for func, args, kwargs in self.items]
  File "/home/circleci/miniconda3/envs/testenv/lib/python3.6/site-packages/sklearn/model_selection/_validation.py", line 528, in _fit_and_score
    estimator.fit(X_train, y_train, **fit_params)
  File "/home/circleci/miniconda3/envs/testenv/lib/python3.6/site-packages/sklearn/linear_model/logistic.py", line 1276, in fit
    solver = _check_solver(self.solver, self.penalty, self.dual)
  File "/home/circleci/miniconda3/envs/testenv/lib/python3.6/site-packages/sklearn/linear_model/logistic.py", line 446, in _check_solver
    "got %s penalty." % (solver, penalty))
ValueError: Solver lbfgs supports only l2 penalties, got l1 penalty.
________________________________________________________________________________

```

**Comment by larsoner** [comment]
Must be a `sklearn` version issue, I'll switch it to `liblinear`, which is listed as the old / 0.20 default in the docstring (which will change to `lbfgs` in 0.22).

**Comment by GaelVaroquaux** [comment]
This is failing, but with our good old download failure, hence I will take the risk to merge, as it does seem an improvement. Thank you!

**Comment by kchawla-pi** [comment]
Thanks for the help @larsoner  !

**Pull request merged**